diff --git a/src/widgets/arc/lv_arc.c b/src/widgets/arc/lv_arc.c
index c7b45d408..964a85427 100644
--- a/src/widgets/arc/lv_arc.c
+++ b/src/widgets/arc/lv_arc.c
@@ -1145,6 +1145,7 @@ static void arc_value_changed_event_cb(lv_event_t * e)
 {
     lv_obj_t * arc = lv_event_get_current_target(e);
     lv_subject_t * subject = lv_event_get_user_data(e);
+    if(subject == NULL) return;

     if(subject->type == LV_SUBJECT_TYPE_INT) {
         lv_subject_set_int(subject, lv_arc_get_value(arc));
